### DISCUSSION TITLE
Update an existing abbreviation

### DISCUSSION THREAD

**Discussion by ealap** [olets/zsh-abbr#87] [Ideas] [answered]
I currently maintain a dotfile where I define aliases and functions used on different machines. However, some of these machines can't support installation of zsh-abbr that's why I stick with defining them as `alias myalias=...`, and then just run `abbr import-aliases` on machines that has it.

My problem is if I update an alias definition, it is not possible to update the corresponding abbreviation using `add` or `import-aliases`, instead the following error is printed:
```
Did not add the global user abbreviation `myalias` because it already exists
```
I need to use `erase` first, then proceed with `add` or `import-aliases`.

Suggestions:
- Make the `--force` option be able to update an existing abbreviation
- Or introduce a new option `--update` or `--overwrite`

**Comment by olets** [comment]
Letting `add --force` and `import-aliases --force` overwrite an existing abbreviation (using the output as `abbr erase`) sounds like a good solution. I won't have much zsh-abbr time for a while; PRs are welcome.

For you specific issue of duplicating your zsh environment across machines, [zsh-bin or xxh](https://github.com/romkatv/zsh-bin#no-seriously-why) might be helpful (I'm not an expert on either).

**Comment by olets** [comment, marked as answer]
https://github.com/olets/zsh-abbr/releases/tag/v5.1.0 is out, with support for `add --force`, `rename --force`, and `import-aliases --force`
